[Sharmaine Teo] iP

sharmaine1028 opened the pull request:
# Duke
> "Your mind is for having ideas, not holding them." - David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Duke frees your mind of having to remember things you need to do. It's
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/sharmaine1028/ip/releases/tag/A-Jar)
2. double-click it
3. add you tasks
4. let is manage you tasks for you 😉

and it is <b>FREE<b>!

_Features:_
- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon)
---
If you are a java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

sharmaine1028 committed:
Add greet, echo, exit function for Level 1

sharmaine1028 committed:
Add level 2 tasks: Add, List

sharmaine1028 committed:
Add task class

sharmaine1028 committed:
Ignore class files

sharmaine1028 committed:
Add mark and unmark function

sharmaine1028 committed:
Add DukeException

sharmaine1028 committed:
Add task class

sharmaine1028 committed:
Add parser to handle commands

sharmaine1028 committed:
Add main class as entry point to application

sharmaine1028 committed:
Add DukeException class

sharmaine1028 committed:
Edit Duke to handle commands

sharmaine1028 committed:
Add unmark task command

sharmaine1028 committed:
Add duke exception class

sharmaine1028 committed:
Add add task command

sharmaine1028 committed:
Add bye command

sharmaine1028 committed:
Add list all taks command

sharmaine1028 committed:
Add command class

sharmaine1028 committed:
Add mark command

sharmaine1028 committed:
Ignore vscode settings and class files

sharmaine1028 committed:
Remove unnecessary files

sharmaine1028 committed:
Edit add command and add printMessage method

sharmaine1028 committed:
Edit imports

sharmaine1028 committed:
Add task, todo, event, deadline classes

sharmaine1028 committed:
Add commands for todo, deadline, event

sharmaine1028 committed:
Move task.java to separate folder

sharmaine1028 committed:
Edit parser to handle, todo, deadline and event

sharmaine1028 committed:
Add javadocs and rearrange files

sharmaine1028 committed:
Automate testing

sharmaine1028 committed:
Add error handling

sharmaine1028 committed:
Add delete task functionality

sharmaine1028 committed:
Implement enums to handle commands

sharmaine1028 committed:
Add delete JUnit test

sharmaine1028 committed:
Refactor code to follow OOP principles

sharmaine1028 committed:
Add storage class

sharmaine1028 committed:
Add Ui class

sharmaine1028 committed:
Edit task classes to handle saving taskslist to storage

sharmaine1028 committed:
Change Command class from functional interface to abstract class

sharmaine1028 committed:
Remove unnecessary files

sharmaine1028 committed:
Integrate Ui, Storage and Parser together in Duke class

sharmaine1028 committed:
Change fields of at and by for Deadline and Event to be LocalDate objects instead of string

sharmaine1028 committed:
Resolve merge conflics from Level-7 and Level-8

sharmaine1028 committed:
Fix bug of mark and unmark commands

sharmaine1028 committed:
Arrange all file into packages

sharmaine1028 committed:
Add gradle support

sharmaine1028 committed:
Fix packages after adding Gradle support

sharmaine1028 committed:
Add test cases for Task package

sharmaine1028 committed:
Fix mainClassName in build.gradle

sharmaine1028 committed:
Add JavaDoc to methods

sharmaine1028 committed:
Add JavaDocs

sharmaine1028 committed:
Revert "Add JavaDoc to methods"

This reverts commit afa166c3892db0cdf993c68963326aacc90a1a01.

sharmaine1028 committed:
Edit document to comply with Java coding standards

sharmaine1028 committed:
Edit method call in TaskTest to comply with Java coding standard

sharmaine1028 committed:
Add find command functionality

sharmaine1028 committed:
Merge branch 'branch-A-JavaDoc'

sharmaine1028 committed:
Merge branch 'branch-A-CodingStandard'

sharmaine1028 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/main/Parser.java

sharmaine1028 committed:
Add JavaDocs and follow coding standards

sharmaine1028 committed:
Add checkstyle

sharmaine1028 committed:
Add a GUI to Duke

sharmaine1028 committed:
Merge branch 'branch-A-CheckStyle'

sharmaine1028 committed:
Merge branch 'branch-Level-10'

sharmaine1028 committed:
Fix checkstyle errors

JJoeYi commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,34 @@
+ * Parent class of all commands with single execute method
+ */
+
+public abstract class Command {
```
I like the way you encapsulate commands into its own abstract class!

JJoeYi commented on src/main/java/duke/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,51 @@
+    }
+
+    /**
+     * Marks command and prints out message to users depending on whether the
+     * command was successful
```
I like the use of JavaDocs here, very helpful

JJoeYi commented on src/main/java/duke/commands/DeleteCommand.java:
```diff
@@ -0,0 +1,51 @@
+     * @param description String representation of task number to be marked
+     * @throws DukeException if user did not type in a correct task number
+     */
+    public DeleteCommand(String description) throws DukeException {
```
I like the naming convention, clearly tells me what the method does without having to read the implementation.

JJoeYi commented on src/main/java/duke/main/CommandManager.java:
```diff
@@ -0,0 +1,16 @@
+/**
+ * CommandManager lists down all possible commands that can be taken by user
+ */
+public enum CommandManager {
```
Great use of enums here.

JJoeYi commented on src/main/java/duke/main/Parser.java:
```diff
@@ -0,0 +1,51 @@
+     */
+    static Command parse(String text) throws DukeException {
+        String command = text.split(" ")[0];
+        switch (CommandManager.valueOf(command)) {
+        case list:
```
Great use of switch-case. Following the indentation guide according to the Java coding standards!

JJoeYi reviewed (commented):
LGTM. 🙏

sharmaine1028 committed:
Add green text box for user and white text box for Duke

sharmaine1028 committed:
Add assertions

sharmaine1028 committed:
Refactor Ui class out

The Ui class prints out messages to
users in the command line.

However,the project has changed from a command line one to
Gui, this function is no longer necessary.

Get response by Duke using getResponse function in Duke

Do it this way so that we can convert response by Duke to Labels in
order to display to user in Gui

Let's delete all occurrences of Ui class in all methods to make code
deater

sharmaine1028 committed:
Refactor commands to separate function and return message

sharmaine1028 committed:
Merge pull request #1 from sharmaine1028/branch-A-Assertions

Add assertions

sharmaine1028 committed:
Fix merge conflicts

sharmaine1028 committed:
Merge pull request #2 from sharmaine1028/branch-A-CodeQuality

Branch a code quality

sharmaine1028 committed:
Fix bug of byecommand not exiting Duke

sharmaine1028 committed:
Add margin to picture and label in dialogBox

sharmaine1028 committed:
Add tag and untag class

sharmaine1028 committed:
Fix bugs in storage and parser due to adding of tag

sharmaine1028 committed:
Add images for the user, chatbox and background

sharmaine1028 committed:
Set resizable to false

sharmaine1028 committed:
Revert "Set resizable to false"

This reverts commit f46bb35366a944c12b9e6b7d90ca4ceeac608f58.

sharmaine1028 committed:
Fix bug in storage -- saving same task multiple times

sharmaine1028 committed:
Tweak GUI to match asymmetric conversation and highlight errors

sharmaine1028 committed:
Merge branch 'master' of https://github.com/sharmaine1028/ip

sharmaine1028 committed:
Add personality in theme with Red Queen and Knave of hearts

sharmaine1028 committed:
Fix minor bugs such as spacings

sharmaine1028 committed:
Edit user guide

sharmaine1028 committed:
Fix checkstyle errors

sharmaine1028 committed:
Rename Jar file

sharmaine1028 committed:
Edit User Guide

sharmaine1028 committed:
Fix bug of unknown command not returning a red message
